### PR TITLE
Handle portal app callback URLs

### PR DESCRIPTION
**Pull request by sadilchamishka**
### Proposed changes in this pull request
$subject

### CODE REVIEW

**Review by shashimalcse — APPROVED** [review]

**Review by ThaminduDilshan — APPROVED** [review]

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java [hunk 1]
```diff
@@ -1087,6 +1089,28 @@ public static String resolveOriginUrlFromPlaceholders(String absoluteUrl) throws
...
+     * This method use to replace placeholders with the hostname and port of URLs for the portal apps.
+     *
+     * @param absoluteUrl     The URL which need to resolve from placeholders.
+     * @return The resolved URL from placeholders.
+     * @throws URLBuilderException If any error occurs when building absolute public url without path.
+     */
+    public static String resolveOriginUrlFromPlaceholdersPortalApps(String absoluteUrl, String appName)
+            throws URLBuilderException {
+
+        String basePath = "";
```
Comment (AnuradhaSK) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:1102]: Let's use StringUtils.EMPTY

Comment (sadilchamishka) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java]: Fixed by - https://github.com/wso2/carbon-identity-framework/pull/5656/commits/70758a641e4707c5c1654cf80db79e50e13da48d

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java [hunk 2]
```diff
@@ -1087,6 +1089,28 @@ public static String resolveOriginUrlFromPlaceholders(String absoluteUrl) throws
...
     }
 
+    /**
+     * This method use to replace placeholders with the hostname and port of URLs for the portal apps.
+     *
+     * @param absoluteUrl     The URL which need to resolve from placeholders.
+     * @return The resolved URL from placeholders.
+     * @throws URLBuilderException If any error occurs when building absolute public url without path.
+     */
+    public static String resolveOriginUrlFromPlaceholdersPortalApps(String absoluteUrl, String appName)
```
Comment (AnuradhaSK) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:1099]: Let's make this a generic method, by overloading resolveOriginUrlFromPlaceholders

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java [hunk 3]
```diff
@@ -1087,6 +1089,28 @@ public static String resolveOriginUrlFromPlaceholders(String absoluteUrl) throws
...
+        String basePath = StringUtils.EMPTY;
+        if (ApplicationConstants.CONSOLE_APPLICATION_NAME.equals(appName)) {
+            basePath = IdentityUtil.getProperty(CONSOLE_ACCESS_ORIGIN);
+        } else if (ApplicationConstants.MY_ACCOUNT_APPLICATION_NAME.equals(appName)) {
+            basePath = IdentityUtil.getProperty(MYACCOUNT_ACCESS_ORIGIN);
+        }
+        if (StringUtils.isEmpty(basePath)) {
+            basePath = ServiceURLBuilder.create().build().getAbsolutePublicUrlWithoutPath();
+        }
+        return StringUtils.replace(absoluteUrl, BASE_URL_PLACEHOLDER, basePath);
```
[the remark this answers is not in the document]
Comment (sadilchamishka) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java]: Fixed by - https://github.com/wso2/carbon-identity-framework/pull/5656/commits/9e6fadcb58d087929ff2ebf80b51323c15ae5764

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**